src/commands/code-analyzer/run.ts

```diff
@@ -67,8 +75,58 @@ export default class RunCommand extends SfCommand<void> {
 
 	public async run(): Promise<void> {
 		const parsedFlags = (await this.parse(RunCommand)).flags;
-		const action: RunAction = new RunAction();
-		action.execute(parsedFlags);
+		const dependencies: RunDependencies = this.createDependencies(parsedFlags.view as View, parsedFlags['output-file']);
+		const action: RunAction = RunAction.createAction(dependencies);
+		const runInput: RunInput = {
+			'config-file': parsedFlags['config-file'],
+			'path-start': parsedFlags['path-start'],
+			'rule-selector': parsedFlags['rule-selector'],
+			'workspace': parsedFlags['workspace']
+		};
+		const possibleThreshold = this.convertThresholdToEnum(parsedFlags['severity-threshold']);
+		if (possibleThreshold) {
+			runInput['severity-threshold'] = possibleThreshold;
+		}
+		await action.execute(runInput);
+	}
+
+	protected createDependencies(view: View, outputFiles: string[] = []): RunDependencies {
+		const uxDisplay: UxDisplay = new UxDisplay(this);
+		return {
+			configFactory: new CodeAnalyzerConfigFactoryImpl(),
+			engineFactory: new EnginePluginFactoryImpl(),
+			outputFileWriter: new OutputFileWriterImpl(outputFiles),
+			viewer: view === View.TABLE
+				? new ResultsTableViewer(uxDisplay)
+				: new ResultsDetailViewer(uxDisplay)
+		};
+	}
+
+	private convertThresholdToEnum(threshold: string|undefined): SeverityLevel|undefined {
+		// We could do all sorts of complicated conversion logic, but honestly it's just easier
+		// to do a switch-statement.
+		switch (threshold) {
+			case undefined:
+				return undefined;
```

Comment:
Reasonable. I'll do that.

Reply:
Sorry, I'm late to this. What is the downstream behavior when undefined is returned? Wouldn't it be better for the default threshold to be Info?

Reply:
Jag, undefined just means that the user didn't provide it to the CLI. In this case, code analyzer would just exit as normal instead of existing with a non-zero return value (which is the default behavior).